### PR TITLE
Search Bar Styling and Optimization

### LINKED ISSUE #354: Style All Search Bars
Django-Machina's default search bar styling is much nicer then the search bar styling that currently exists in the project. This issue aims to look into if it's practical to apply the Django-Machina styling search bar styling to all search bars.

### LINKED ISSUE #359: Explore how to make default search box dependent on the current page
This issue aims to explore how to make the default search box be dependent on the page.

### PR DESCRIPTION
**Pull request by Benjoman12**
Per issue #354 I have added the django-machina search bar styling to the nav search bar. Per issue #359 I have made it so that the default selected item in the dropdown is dependent on the current page.

### CODE REVIEW

**Comment by Benjoman12** [comment]
Due to time constraints, this issue relies heavily on PR #358 and thus shouldn't be reviewed until PR #358 is merged

**Review by joshuamonroe — APPROVED** [review]
This looks great! Really appreciate having the search box automatically selected based on the page context now. Glad we were able to work the kinks out of this before review and good job retaining the existing code comments or replacing them where things have changed.

File: src/templates/base.html [hunk 1]
```diff
@@ -113,18 +113,41 @@
...
+                              {% if request.path == users_search_url %}selected{% endif %}>Users</option>
+                      <option value="forums"
+                              {% if request.path == forum_url or request.path == forum_search_url %}selected{% endif %}>
+                        Forums
+                      </option>
+                      <!-- Add more options here -->
+                    </select>
+                  </div>
+                  <div>
+                    <!-- Name must remain as "q" as django-machina expects that name -->
```
Comment (majorsylvie) [inline src/templates/base.html:143]: Thank you for this code comment!

File: src/templates/base.html [hunk 2]
```diff
@@ -256,13 +279,11 @@
                 source: []  // placeholder; replace with list of forums
               });
             }
+            advancedSearchButton.classList.toggle('d-none', query_type.value !== "forums"); // hide advanced search button if not searching forums
```
Comment (majorsylvie) [inline src/templates/base.html:282]: Nice clean way to accomplish this functionality. Thank you for code coment

**Review by majorsylvie — APPROVED** [review]
This looks awesome! I love how the dropdown, search, and advanced search button are all the same style. Feels significantly more polished than before!

Similarly, having the dropdown default change depending on location is super nice for UX. 

Great work! Let's get this merged.

**Comment by Benjoman12** [comment]
Closed and merged to dev without significant revision. Thanks to @joshuamonroe for reviewing the PR.

**Comment by majorsylvie** [comment]
Issue Score: Excellent

Comments:
The title is too generic and could relate to other teams.

**Pull request merged**